cmake/KokkosConfigCommon.cmake.in

```diff
@@ -203,6 +203,45 @@ int main()
   set(${_VAR} ${_RET} PARENT_SCOPE)
 endfunction()
 
+# this function checks whether the current CXX compiler supports building Kokkos
+#
+#       RESULT      --> result (bool)
+#
+function(kokkos_cxx_compiler_is_compatible)
+  cmake_parse_arguments(INP "" "RESULT" "" ${ARGN})
+
+  file(
+    WRITE ${PROJECT_BINARY_DIR}/compile_tests/compiles_kokkos.cpp
+    "
+// include Kokkos_Core to check if header is working with the compiler
+#include <Kokkos_Core.hpp>
+
+int main()
+{
+Kokkos::initialize();
+{ // instantiate a view and a parallel_for to check requirements for template instantiation
+    Kokkos::View<int*> V(\"V\", 10);
+    Kokkos::parallel_for(\"increment\", 10, KOKKOS_LAMBDA ( int j) {
+        ++V(j);
+    });
+    Kokkos::fence();
+}
+Kokkos::finalize();
+return EXIT_SUCCESS;
+}
+"
+  )
+
+  try_compile(
+    COMPILER_CAN_COMPILE_KOKKOS ${PROJECT_BINARY_DIR}/compile_tests
```

Comment:
The result variable will be cached unless you specify `NO_CACHE` so there is no reason to propagate the result explicitly at this point.

Reply:
by switching to CMAKE_CXX_SOURCE_COMPILES this is taken care of